### ISSUE TITLE
The conditional check 'XX <= 15' failed. The error was: Conditional is marked as unsafe, and cannot be evaluated.

### ISSUE THREAD

**Issue by audiolomb** [sap-linuxlab/community.sap_install#555]
### Summary

When running sap_general_preconfigure, getting the error above.

### Issue Type

Bug Report

### Component Name

sap_general_preconfigure

### Ansible Version

```console (paste below)
$ ansible --version

```
ansible [core 2.16.2]
  config file = /home/user/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.11/site-packages/ansible
  ansible collection location = /home/user/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.11.5 (main, Sep  7 2023, 00:00:00) [GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] (/usr/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True

### Community.sap_install Version

```console (paste below)
$ ansible-galaxy collection list community.sap_install
community.sap_install 1.3.2  

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```
COLLECTIONS_PATHS(env: ANSIBLE_COLLECTIONS_PATH) = ['/home/user/ansible/collections']
CONFIG_FILE() = /home/user/ansible/ansible.cfg
DEFAULT_ACTION_PLUGIN_PATH(/home/user/ansible/ansible.cfg) = ['/usr/share/ansible/plugins/action']
DEFAULT_FORKS(/home/user/ansible/ansible.cfg) = 50
DEFAULT_REMOTE_USER(/home/user/ansible/ansible.cfg) = ansible
DEFAULT_ROLES_PATH(env: ANSIBLE_ROLES_PATH) = ['/home/user/ansible/roles']
DEFAULT_VAULT_PASSWORD_FILE(/home/user/ansible/ansible.cfg) = /home/user/vault-pass
HOST_KEY_CHECKING(/home/user/ansible/ansible.cfg) = False

### OS / Environment

RHEL 9 (VMware)

### Steps to Reproduce

```yaml (paste below)

```
      - import_role:
          name: community.sap_install.sap_general_preconfigure

VARS:
sap_general_preconfigure_reboot_ok: true
sap_general_preconfigure_selinux_state: disabled
sap_general_preconfigure_modify_selinux_labels: false
sap_general_preconfigure_max_hostname_length: 15

### Expected Results

I expect the task "community.sap_install.sap_general_preconfigure : Ensure that the length of the hostname is not longer than 'sap_general_preconfigure_max_hostname_length'" to complete. Instead Ansible crashes

### Actual Results

```console (paste below)


TASK [community.sap_install.sap_general_preconfigure : Ensure that the length of the hostname is not longer than 'sap_general_preconfigure_max_hostname_length'] ***
task path: /home/user/ansible/playbook/prod/collections/ansible_collections/community/sap_install/roles/sap_general_preconfigure/tasks/RedHat/generic/configure-hostname.yml:51
fatal: [hostname.domain]: FAILED! => {
    "msg": "The conditional check '10 <= 15' failed. The error was: Conditional is marked as unsafe, and cannot be evaluated."
}

PLAY RECAP ***********************************************************************************************************************************************
hostname.domain     : ok=74   changed=2    unreachable=0    failed=1    skipped=68   rescued=0    ignored=0   
```

**Comment by berndfinger** [comment]
Hi @audiolomb audiolomb Many thanks for reporting this. I have just created https://github.com/sap-linuxlab/community.sap_install/issues/556 to broaden the scope.

**Comment by berndfinger** [comment]
See also https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_9.html#playbook .

**Comment by berndfinger** [comment]
Fixed in commit 738efaa2b5d63af884c6a4f4a1df961079da0b98, now part of collection release 1.3.3.
